airflow/providers/google/cloud/hooks/bigquery.py

```diff
@@ -2442,10 +2442,17 @@ def next(self) -> Union[List, None]:
             if self.all_pages_loaded:
                 return None
 
-            query_results = (self.service.jobs().getQueryResults(
-                projectId=self.project_id,
-                jobId=self.job_id,
-                pageToken=self.page_token).execute(num_retries=self.num_retries))
+            if self.location:
+                query_results = (self.service.jobs().getQueryResults(
+                    projectId=self.project_id,
+                    jobId=self.job_id,
+                    location=self.location,
+                    pageToken=self.page_token).execute(num_retries=self.num_retries))
+            else:
+                query_results = (self.service.jobs().getQueryResults(
+                    projectId=self.project_id,
+                    jobId=self.job_id,
+                    pageToken=self.page_token).execute(num_retries=self.num_retries))
```

Comment:
Would passing `location=None` work?

Reply:
Not sure, I noticed this highly verbose pattern in the rest of the BigQuery related classes (just look at any other mention of `location` in `big_query_hook.py`) so I just followed it's style. I will check it and update if needed.

Reply:
Yep, that seems to work in this case!

Reply:
@edejong thanks!